server/src/main/java/org/opensearch/index/store/lockmanager/RemoteStoreLockManager.java

```diff
@@ -38,4 +38,12 @@ public interface RemoteStoreLockManager {
      * @throws IOException throws exception in case there is a problem in checking if a given file is locked or not.
      */
     Boolean isAcquired(LockInfo lockInfo) throws IOException;
+
+    /**
+     *
+     * @param originalLockInfo lock info instance for original lock.
+     * @param clonedLockInfo lock info instance for which lock needs to be cloned.
+     * @throws IOException throws IOException if originalResource itself do not have any lock.
+     */
+    void cloneLock(LockInfo originalLockInfo, LockInfo clonedLockInfo) throws IOException;
```

Comment:
Not convinced this is the right way to handle locks. Lets avoid these complexities this might lead to a deadlock?

Reply:
Yes, as suggested by @sohami [here](https://github.com/opensearch-project/OpenSearch/pull/7496/#discussion_r1223564993), updated the java doc to call out the race condition scenario. Until we have a generic fix on the lock manager side, consumers must enforce the thread safety. Here, the snapshot layer takes care of it by having checks before initiating the clone operation that the source snapshot is not being deleted and hence the files that we need during cloning would indeed be present.